optax/_src/alias.py

```diff
@@ -1120,6 +1120,66 @@ def noisy_sgd(
   )
 
 
+def sign_sgd(
+    learning_rate: base.ScalarOrSchedule,
+) -> base.GradientTransformation:
+  r"""A variant of SGD that updates using only the signs of the gradient
+  components.
+
+  SignSGD is a variant of SGD that uses the signs of the gradient components in
+  the update, not their actual values. The update :math:`u_t` is modified as
+  follows:
+
+  .. math::
+    u_t \leftarrow -\alpha_t\, \text{sign}\,(g_t),
+
+  SGD variants that use only the signs of the gradient update have historically
+  been used since RProp, with modern forms including RMSProp, Adam, and Lion.
+  SignSGD uses only the signs of the gradient update, and has been shown to
+  perform comparably with adaptive gradient methods like Adam. SignSGD also
+  enables significant gradient compression, substantially reducing the
+  bottleneck imposed by communicating gradients when distributing learning
+  across multiple workers.
+
+  Examples:
+    >>> import optax
+    >>> import jax
+    >>> import jax.numpy as jnp
+    >>> def f(x): return jnp.sum(x ** 2)  # simple quadratic function
+    >>> solver = optax.sign_sgd(learning_rate=0.003)
+    >>> params = jnp.array([1., 2., 3.])
+    >>> print('Objective function: ', f(params))
+    Objective function:  14.0
+    >>> opt_state = solver.init(params)
+    >>> for _ in range(5):
+    ...  grad = jax.grad(f)(params)
+    ...  updates, opt_state = solver.update(grad, opt_state, params)
+    ...  params = optax.apply_updates(params, updates)
+    ...  print('Objective function: {:.2E}'.format(f(params)))
+    Objective function: 1.40E+01
+    Objective function: 1.39E+01
+    Objective function: 1.39E+01
+    Objective function: 1.39E+01
+    Objective function: 1.38E+01
+
+    
+  References:
+    Bernstein et al, `signSGD: Compressed Optimisation for Non-Convex Problems 
+    <https://arxiv.org/abs/1802.04434>`_, 2018
+
```

Comment:
```suggestion
Balles et al.`The Geometry of Sign Gradient Descent <https://arxiv.org/abs/2002.08056>`, 2020
```

(I'm biased since I'm an author on that paper but I think it gives a complementary viewpoint to the other referece, which covers mostly the distributed part)